src/iron.rs

```diff
@@ -58,7 +75,13 @@ impl<H: Handler> Iron<H> {
         let SocketAddr { ip, port } = sock_addr.clone();
         self.addr = Some(sock_addr);
 
-        Ok(try!(Server::http(ip, port).listen_threads(self, threads)))
+        let server = match self.security {
+            Security::Insecure => Server::http(ip, port),
+            Security::Secure { ref certificate, ref key } =>
+                Server::https(ip, port, certificate.clone(), key.clone())
```

Comment:
It seems like it is incorrect for the `Iron` instance to own the certificate and key here. Maybe the protocol should instead be passed as an argument to `listen` and `listen_with`?